types/types.go

```diff
@@ -59,7 +59,13 @@ func NewWarpBlockInfo(logger logging.Logger, header *types.Header, ethClient eth
 			})
 			return err
 		}
-		err = utils.WithRetriesTimeout(logger, operation, utils.DefaultRPCTimeout, "get warp logs from block")
+
+		// We increase the timeout here to reduce the chance of hitting a race condition
+		// where the block header is received via websocket subscription before the block's
+		// logs are available via RPC. This is a known behavior in EVM nodes due to
+		// asynchronous log/index processing after a block becomes canonical.
+		timeout := utils.DefaultRPCTimeout * 2
```

Comment:
I would be okay setting this even higher. If we got the header, the logs should be available eventually, and we have no recovery mechanism either, so I think waiting somewhere on the order of ~30s would still be appropriate